### PR TITLE
bpo-46044: Annotate deprecated sdists formats

### PR DESCRIPTION
**Pull request by Carreau**
While this page have deprecated informations it is still heavily index
by Google.

Discussed on twitter: https://twitter.com/brettsky/status/1469465729082662916

https://bugs.python.org/issue46044

Automerge-Triggered-By: GH:brettcannon

### CODE REVIEW

File: Doc/distutils/sourcedist.rst [hunk 1]
```diff
@@ -61,6 +61,9 @@ Notes:
 (4)
    requires the :program:`compress` program. Notice that this format is now
    pending for deprecation and will be removed in the future versions of Python.
+(5)
+  Those have been deprecated by `Pep 527 <https://www.python.org/dev/peps/pep-0527/>`_,
+  and PyPI only accepts ``.zip`` and ``.tar.gz`` file.
```
Comment (brettcannon) [inline Doc/distutils/sourcedist.rst]:
```suggestion
  deprecated by `PEP 527 <https://www.python.org/dev/peps/pep-0527/>`_;
  `PyPI <https://pypi.org>`_ only accepts ``.zip`` and ``.tar.gz`` files.
```

**Review by brettcannon — CHANGES_REQUESTED** [review]
Slight grammar tweak and adding a link to PyPI.

**Comment by Carreau** [comment]
Done. Thanks !

**Comment by brettcannon** [comment]
@Carreau for future reference, to tell the bot to update the state of the PR, you need to paste in the phrase, I have made the requested changes; please review again .

**Review by brettcannon — APPROVED** [review]

File: Doc/distutils/sourcedist.rst [hunk 2]
```diff
@@ -23,25 +23,25 @@ option, for example::
...
+| ``gztar`` | gzip'ed tar file        | \(2)        |
+|           | (:file:`.tar.gz`)       |             |
++-----------+-------------------------+-------------+
+| ``bztar`` | bzip2'ed tar file       | \(5)        |
+|           | (:file:`.tar.bz2`)      |             |
++-----------+-------------------------+-------------+
+| ``xztar`` | xz'ed tar file          | \(5)        |
+|           | (:file:`.tar.xz`)       |             |
++-----------+-------------------------+-------------+
+| ``ztar``  | compressed tar file     | \(5)        |
```
Comment (bluetech) [inline Doc/distutils/sourcedist.rst:40]: Note 4 should probably remain here? Or else it should be removed.

Comment (Carreau) [reply Doc/distutils/sourcedist.rst:40]: Hum sorry, not sure how it got removed, likely a repeat command with vim that I messed up.

**Pull request merged**